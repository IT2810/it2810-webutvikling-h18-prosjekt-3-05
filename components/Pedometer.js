import Expo from "expo";
import React from "react";
import { Pedometer } from "expo";
import { StyleSheet, Text, View } from "react-native";

// https://docs.expo.io/versions/v30.0.0/sdk/pedometer
export default class StepCounter extends React.Component {
    state = {
        isPedometerAvaliable: "Check",
        pastSteps: 0,
        currentSteps: 0
    };
    
  componentDidMount() {
    this._subscribe();
  }

  componentWillUnmount() {
    this._unsubscribe();
  }

  _subscribe = () => {
    this._subscription = Pedometer.watchStepCount(result => {
      this.setState({
        currentSteps: result.steps
      })
    });

    Pedometer.isAvailableAsync().then(
      result => {
        this.setState({
          isPedometerAvailable: String(result)
        });
      },
      error => {
        this.setState({
          isPedometerAvailable: "Pedometer is not avaliable: " + error
        });
      }
    );

    const end = new Date();
    const start = new Date();
    start.setDate(end.getDate() - 1);
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ pastSteps: result.steps });
      },
      error => {
        this.setState({
          pastSteps: "Could not get stepCount: " + error
        });
      }
    );
  };
// This function will not run when reloading the expo client and cause an error:
// "Already managing a GoogleApiClient with id 0"
// This issue will only occur when using the expo client
  _unsubscribe = () => {
    this._subscription && this._subscription.remove();
    this._subscription = null;
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>
          Pedometer is avaliable: {this.state.isPedometerAvailable}
        </Text>
        <Text style={styles.text}>
          Steps taken in the last 24 hours: {this.state.pastSteps}
        </Text>
        <Text style={styles.text}>Walk! And watch this go up: {this.state.currentSteps}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#01194f',
    },
    text: {
        fontSize: 17,
        color: '#2e78b7'
    }
});

Expo.registerRootComponent(StepCounter);
