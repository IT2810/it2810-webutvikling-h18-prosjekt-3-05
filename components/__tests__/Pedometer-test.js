import React from 'react';
import Pedometer from '../Pedometer';
import renderer from 'react-test-renderer';

it('should render corrrectly', () => {
  const component = renderer.create(<Pedometer />)
  const tree = renderer.create(<Pedometer />).toJSON();
  expect (tree).toMatchSnapshot();
})

/*it('should watchStepCount when _subscribe is called', () => {
  const component = renderer.create(<Pedometer />)
  const instance = component.root.instance
  const state = instance.state
  const mockSetState = jest.fn()
  instance.watchStepCount = mockSetState
  console.log(state)
  instance.componentDidMount()
  expect(mockSetState).toHaveBeenCalled()
})*/
