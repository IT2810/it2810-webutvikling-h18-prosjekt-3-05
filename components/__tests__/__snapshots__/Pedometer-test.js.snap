// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render corrrectly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#01194f",
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#2e78b7",
        "fontSize": 17,
      }
    }
  >
    Pedometer is avaliable: 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#2e78b7",
        "fontSize": 17,
      }
    }
  >
    Steps taken in the last 24 hours: 
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#2e78b7",
        "fontSize": 17,
      }
    }
  >
    Walk! And watch this go up: 
    0
  </Text>
</View>
`;
